tests/sklearn/test_sklearn_autolog.py

```diff
@@ -468,16 +528,38 @@ def throwing_score(X, y=None, sample_weight=None):  # pylint: disable=unused-arg
 
     model.score = throwing_score
 
-    with mlflow.start_run() as run, mock.patch("mlflow.sklearn._logger.warning") as mock_warning:
+    with mlflow.start_run(), mock.patch("mlflow.sklearn._logger.warning") as mock_warning:
         model.fit(*get_iris())
         mock_warning.assert_called_once()
         mock_warning.called_once_with(
             "KMeans.score failed. The 'training_score' metric will not be recorded. "
             "Scoring error: EXCEPTION"
         )
 
-    metrics = get_run_data(run.info.run_id)[1]
-    assert metrics == {}
+
+def test_autolog_emits_warning_message_when_metric_fails():
+    """
+    Take precision_score metric from SVC as an example to test metric logging failure
+    :return: NULL
+    """
+    mlflow.sklearn.autolog()
+
+    model = sklearn.svm.SVC()
+
+    @functools.wraps(model.score)
+    def throwing_score(y_true, y_pred):  # pylint: disable=unused-argument
+        raise Exception("EXCEPTION")
+
+    sklearn.metrics.precision_score = throwing_score
```

Comment:
The reason I patched `model.score` this way was to preserve its signature (for [this line](https://github.com/mlflow/mlflow/blob/master/mlflow/sklearn/utils.py#L90)). sorry I should have put a comment on this. If you don't have to preserve a signature, you can just use `mock.patch` with `side_effect`.